packages/webapp/components/layouts/AccountLayout/Security/index.tsx

```diff
@@ -0,0 +1,142 @@
+import { getProviderMapClone } from '@dailydotdev/shared/src/components/auth/common';
+import ProviderButton from '@dailydotdev/shared/src/components/auth/ProviderButton';
+import { Button } from '@dailydotdev/shared/src/components/buttons/Button';
+import LockIcon from '@dailydotdev/shared/src/components/icons/Lock';
+import MailIcon from '@dailydotdev/shared/src/components/icons/Mail';
+import AccountDangerZone from '@dailydotdev/shared/src/components/profile/AccountDangerZone';
+import { Overlay } from '@dailydotdev/shared/src/components/utilities';
+import AuthContext from '@dailydotdev/shared/src/contexts/AuthContext';
+import React, { ReactElement, useContext, useState } from 'react';
+import AccountContentSection from '../AccountContentSection';
+import { AccountPageContainer } from '../AccountPageContainer';
+import {
+  AccountSecurityDisplay as Display,
+  AccountTextField,
+  OverlayContainer,
+  OverlayText,
+} from '../common';
+import EmailSentSection from '../EmailSentSection';
+import AccountLoginSection from './AccountLoginSection';
+
+const socialProvider = getProviderMapClone();
+socialProvider.gitHub.style = { backgroundColor: '#383C47' };
+socialProvider.apple.style = { backgroundColor: '#404551' };
+const providers = Object.values(socialProvider);
+
+interface AccountSecurityDefaultProps {
+  isEmailSent?: boolean;
+  onSwitchDisplay: (display: Display) => void;
+}
+
+function AccountSecurityDefault({
+  isEmailSent,
+  onSwitchDisplay,
+}: AccountSecurityDefaultProps): ReactElement {
+  const { user } = useContext(AuthContext);
+  const [email, setEmail] = useState<string>();
```

Comment:
Here we have `string` type, but initial value is `undefined`, my suggestion is to set initial value to empty string, so only string are used for this state

Reply:
I had the same scenario back then. I am also into using an empty string. Though from a review before, we used null.

@rebelchris what do you think we should go for moving forward with most (if not all) situations like this?`null`, `undefined`, or an empty string?

![image](https://user-images.githubusercontent.com/13744167/183432484-1e7e2018-484a-430b-9c03-1401efd270d3.png)

Reply:
We do quite a lot of checks on nullable (most of the time) so I personally also think null is the way to go.
Especially when strong typed with the <string> before it.

But love to hear if there are more benefits to using empty string compared to null?

Reply:
I only prefer the empty string to infer the type, but I am cool with using null as we've been using it in the past, and being `null` fits the context.